tests/python_tests/test_rag.py

```diff
@@ -0,0 +1,211 @@
+
+
+@pytest.fixture(scope="class", autouse=True)
+def run_gc_after_test():
+    """
+    Fixture to run garbage collection after each test class.
+    This is a workaround to minimize memory consumption during tests and allow the use of less powerful CI runners.
+    """
+    yield
+    gc.collect()
+
+
+@pytest.fixture(scope="module")
+def dataset_documents(chunk_size=200):
+    print("dataset_documents")
+    return [
+        TEXT_DATASET[i : i + chunk_size]
+        for i in range(0, len(TEXT_DATASET), chunk_size)
+    ]
+
+
+def run_genai(
+    models_path: Path,
+    documents: list[str],
+    config: TextEmbeddingPipeline.Config | None = None,
+    task: Literal["embed_documents", "embed_query"] = "embed_documents",
+):
+    if not config:
+        config = TextEmbeddingPipeline.Config()
+
+    pipeline = TextEmbeddingPipeline(models_path, "CPU", config)
+
+    if task == "embed_documents":
+        return pipeline.embed_documents(documents)
+    else:
+        return pipeline.embed_query(documents[0])
+
+
+def run_langchain(
+    models_path: Path,
+    documents: list[str],
+    config: TextEmbeddingPipeline.Config | None = None,
+    task: Literal["embed_documents", "embed_query"] = "embed_documents",
+):
+    if not config:
+        config = TextEmbeddingPipeline.Config()
+
+    encode_kwargs = {
+        "normalize_embeddings": config.normalize,
+        # batch size affects the result
+        "batch_size": len(documents),
+    }
+    if config.pooling_type == TextEmbeddingPipeline.PoolingType.MEAN:
+        encode_kwargs["mean_pooling"] = True
+
+    ov_embeddings = OpenVINOBgeEmbeddings(
+        model_name_or_path=str(models_path),
+        model_kwargs={"device": "CPU"},
+        encode_kwargs=encode_kwargs,
+    )
+
+    # align instructions
+    ov_embeddings.embed_instruction = config.embed_instruction or ""
+    ov_embeddings.query_instruction = config.query_instruction or ""
+
+    if task == "embed_documents":
+        return ov_embeddings.embed_documents(documents)
+    else:
+        return ov_embeddings.embed_query(documents[0])
+
+
+def run_pipeline_with_ref(
+    models_path: Path,
+    documents: list[str],
+    config: TextEmbeddingPipeline.Config | None = None,
+    task: Literal["embed_documents", "embed_query"] = "embed_documents",
+):
+    genai_result = run_genai(models_path, documents, config, task)
+    langchain_result = run_langchain(models_path, documents, config, task)
+
+    np_genai_result = np.array(genai_result)
+    np_langchain_result = np.array(langchain_result)
+
+    max_error = np.abs(np_genai_result - np_langchain_result).max()
+    print(f"Max error: {max_error}")
+    assert (
+        np.abs(np_genai_result - np_langchain_result).max() < 1e-6
+    ), f"Max error: {max_error}"
+
+
+@pytest.mark.parametrize(
+    "download_and_convert_embeddings_models",
+    [{"model_id": "BAAI/bge-small-en-v1.5"}],
+    indirect=True,
+)
```

Comment:
You can pass just a list of `model_id`, so in a fixture it will be:
```python
@pytest.fixture()
def download_and_convert_embeddings_models(request):
    model_id = request.param
    return _download_and_convert_model(model_id, OVModelForFeatureExtraction)
```